### ISSUE TITLE
'sam local invoke' fails while attempting to pull Lambda layer

### ISSUE THREAD

**Issue by sivarosh** [awslabs/aws-sam-cli#1021]
### Description

When attempting to use the 'sam local invoke' command with my SAM template which consists of a simple serverless function with a Lambda layer, the command invocation failed with the error :

`WindowsError: [Error 2] The system cannot find the file specified: 'C:\\Users\\Administrator\\AppData\\Roaming\\AWS SAM\\layers-pkg\\demo_layer-1-1e6fe5e501'`

In the above scenario, I was using the Python2 package installations of the AWS CLI and the SAM CLI. However, while attempting the same from the Python3 package installations of the AWS CLI and the SAM CLI, the `sam local invoke` command works without any issues.

### Steps to reproduce

1. Initialize a simple SAM Application : `sam init -r python` and add a lambda layer ARN already created before.

```
AWSTemplateFormatVersion: '2010-09-09'
Transform: AWS::Serverless-2016-10-31
Description: >
    sam-app

Resources:
    HelloWorldFunction:
        Type: AWS::Serverless::Function
        Properties:
            CodeUri: hello_world/
            Handler: app.lambda_handler
            Runtime: python3.7
            Layers:
              - arn:aws:lambda:<REGION>:<ACCOUNT_ID>:layer:demo_layer:1

```
2. Run the command `sam local invoke -e event.json --debug`. 
### Observed result

Command execution fails with the error `WindowsError: [Error 2] The system cannot find the file specified: 'C:\\Users\\Administrator\\AppData\\Roaming\\AWS SAM\\layers-pkg\\demo_layer-1-1e6fe5e501'`

Here is the debug output :

```
PS C:\Users\Administrator\sam-test-2\sam-app> sam local invoke -e .\event.json --debug
2019-02-22 04:20:06 Using SAM Template at C:\Users\Administrator\sam-test-2\sam-app\template.yaml
2019-02-22 04:20:06 Changing event name from creating-client-class.iot-data to creating-client-class.iot-data-plane
2019-02-22 04:20:06 Changing event name from before-call.apigateway to before-call.api-gateway
2019-02-22 04:20:06 Changing event name from request-created.machinelearning.Predict to request-created.machine-learning.Predict
2019-02-22 04:20:06 Changing event name from before-parameter-build.autoscaling.CreateLaunchConfiguration to before-parameter-build.auto-scaling.CreateLaunchConfiguration
2019-02-22 04:20:06 Changing event name from before-parameter-build.route53 to before-parameter-build.route-53
2019-02-22 04:20:06 Changing event name from request-created.cloudsearchdomain.Search to request-created.cloudsearch-domain.Search
2019-02-22 04:20:06 Changing event name from docs.*.autoscaling.CreateLaunchConfiguration.complete-section to docs.*.auto-scaling.CreateLaunchConfiguration.complete-section
2019-02-22 04:20:06 Changing event name from before-parameter-build.cloudsearchdomain.Search to before-parameter-build.cloudsearch-domain.Search
2019-02-22 04:20:06 Changing event name from docs.*.cloudsearchdomain.Search.complete-section to docs.*.cloudsearch-domain.Search.complete-section
2019-02-22 04:20:06 Changing event name from before-parameter-build.logs.CreateExportTask to before-parameter-build.cloudwatch-logs.CreateExportTask
2019-02-22 04:20:06 Changing event name from docs.*.logs.CreateExportTask.complete-section to docs.*.cloudwatch-logs.CreateExportTask.complete-section
2019-02-22 04:20:06 Changing event name from creating-client-class.iot-data to creating-client-class.iot-data-plane
2019-02-22 04:20:06 Changing event name from before-call.apigateway to before-call.api-gateway
2019-02-22 04:20:06 Changing event name from request-created.machinelearning.Predict to request-created.machine-learning.Predict
2019-02-22 04:20:06 Changing event name from before-parameter-build.autoscaling.CreateLaunchConfiguration to before-parameter-build.auto-scaling.CreateLaunchConfiguration
2019-02-22 04:20:06 Changing event name from before-parameter-build.route53 to before-parameter-build.route-53
2019-02-22 04:20:06 Changing event name from request-created.cloudsearchdomain.Search to request-created.cloudsearch-domain.Search
2019-02-22 04:20:06 Changing event name from docs.*.autoscaling.CreateLaunchConfiguration.complete-section to docs.*.auto-scaling.CreateLaunchConfiguration.complete-section
2019-02-22 04:20:06 Changing event name from before-parameter-build.cloudsearchdomain.Search to before-parameter-build.cloudsearch-domain.Search
2019-02-22 04:20:06 Changing event name from docs.*.cloudsearchdomain.Search.complete-section to docs.*.cloudsearch-domain.Search.complete-section
2019-02-22 04:20:06 Changing event name from before-parameter-build.logs.CreateExportTask to before-parameter-build.cloudwatch-logs.CreateExportTask
2019-02-22 04:20:06 Changing event name from docs.*.logs.CreateExportTask.complete-section to docs.*.cloudwatch-logs.CreateExportTask.complete-section
2019-02-22 04:20:06 local invoke command is called
2019-02-22 04:20:06 Looking for credentials via: env
2019-02-22 04:20:06 Looking for credentials via: assume-role
2019-02-22 04:20:06 Looking for credentials via: shared-credentials-file
2019-02-22 04:20:06 Found credentials in shared credentials file: ~/.aws/credentials
2019-02-22 04:20:06 Loading JSON file: c:\python27\lib\site-packages\botocore\data\endpoints.json
2019-02-22 04:20:06 Event choose-service-name: calling handler <function handle_service_name_alias at 0x0000000003AE14A8>
2019-02-22 04:20:07 Loading JSON file: c:\python27\lib\site-packages\botocore\data\serverlessrepo\2017-09-08\service-2.json
2019-02-22 04:20:07 Event creating-client-class.serverlessapplicationrepository: calling handler <function add_generate_presigned_url at 0x0000000003A4E6D8>
2019-02-22 04:20:07 The s3 config key is not a dictionary type, ignoring its value of: None
2019-02-22 04:20:07 Setting serverlessrepo timeout as (60, 60)
2019-02-22 04:20:07 Loading JSON file: c:\python27\lib\site-packages\botocore\data\_retry.json
2019-02-22 04:20:07 Registering retry handlers for service: serverlessrepo
2019-02-22 04:20:07 No Parameters detected in the template
2019-02-22 04:20:07 3 resources found in the template
2019-02-22 04:20:07 Found Serverless function with name='HelloWorldFunction' and CodeUri='hello_world/'
2019-02-22 04:20:07 Trying paths: ['C:\\Users\\Administrator\\.docker\\config.json', 'C:\\Users\\Administrator\\.dockercfg']
2019-02-22 04:20:07 No config file found
2019-02-22 04:20:07 Trying paths: ['C:\\Users\\Administrator\\.docker\\config.json', 'C:\\Users\\Administrator\\.dockercfg']
2019-02-22 04:20:07 No config file found
2019-02-22 04:20:07 http://localhost:None "GET /v1.35/_ping HTTP/1.1" 200 2
2019-02-22 04:20:07 Event choose-service-name: calling handler <function handle_service_name_alias at 0x0000000003AE14A8>
2019-02-22 04:20:07 Loading JSON file: c:\python27\lib\site-packages\botocore\data\lambda\2015-03-31\service-2.json
2019-02-22 04:20:07 Event creating-client-class.lambda: calling handler <function add_generate_presigned_url at 0x0000000003A4E6D8>
2019-02-22 04:20:07 The s3 config key is not a dictionary type, ignoring its value of: None
2019-02-22 04:20:07 Setting lambda timeout as (60, 60)
2019-02-22 04:20:07 Registering retry handlers for service: lambda
2019-02-22 04:20:07 Trying paths: ['C:\\Users\\Administrator\\.docker\\config.json', 'C:\\Users\\Administrator\\.dockercfg']
2019-02-22 04:20:07 No config file found
2019-02-22 04:20:07 Trying paths: ['C:\\Users\\Administrator\\.docker\\config.json', 'C:\\Users\\Administrator\\.dockercfg']
2019-02-22 04:20:07 No config file found
2019-02-22 04:20:07 Found one Lambda function with name 'HelloWorldFunction'
2019-02-22 04:20:07 Invoking app.lambda_handler (python3.7)
2019-02-22 04:20:07 No environment variables found for function 'HelloWorldFunction'
2019-02-22 04:20:07 Environment variables overrides data is standard format
2019-02-22 04:20:07 Loading AWS credentials from session with profile 'default'
2019-02-22 04:20:07 Resolving code path. Cwd=C:\Users\Administrator\sam-test-2\sam-app, CodeUri=hello_world/
2019-02-22 04:20:07 Resolved absolute path to code is C:\Users\Administrator\sam-test-2\sam-app\hello_world
2019-02-22 04:20:07 Code C:\Users\Administrator\sam-test-2\sam-app\hello_world is not a zip/jar file
Traceback (most recent call last):
  File "c:\python27\lib\runpy.py", line 174, in _run_module_as_main
    "__main__", fname, loader, pkg_name)
  File "c:\python27\lib\runpy.py", line 72, in _run_code
    exec code in run_globals
  File "C:\Python27\Scripts\sam.exe\__main__.py", line 9, in <module>
  File "c:\python27\lib\site-packages\click\core.py", line 722, in __call__
    return self.main(*args, **kwargs)
  File "c:\python27\lib\site-packages\click\core.py", line 697, in main
    rv = self.invoke(ctx)
  File "c:\python27\lib\site-packages\click\core.py", line 1066, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "c:\python27\lib\site-packages\click\core.py", line 1066, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "c:\python27\lib\site-packages\click\core.py", line 895, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "c:\python27\lib\site-packages\click\core.py", line 535, in invoke
    return callback(*args, **kwargs)
  File "c:\python27\lib\site-packages\click\decorators.py", line 64, in new_func
    return ctx.invoke(f, obj, *args[1:], **kwargs)
  File "c:\python27\lib\site-packages\click\core.py", line 535, in invoke
    return callback(*args, **kwargs)
  File "c:\python27\lib\site-packages\samcli\commands\local\invoke\cli.py", line 56, in cli
    parameter_overrides)  # pragma: no cover
  File "c:\python27\lib\site-packages\samcli\commands\local\invoke\cli.py", line 99, in do_cli
    stderr=context.stderr)
  File "c:\python27\lib\site-packages\samcli\commands\local\lib\local_lambda.py", line 89, in invoke
    self.local_runtime.invoke(config, event, debug_context=self.debug_context, stdout=stdout, stderr=stderr)
  File "c:\python27\lib\site-packages\samcli\local\lambdafn\runtime.py", line 81, in invoke
    debug_options=debug_context)
  File "c:\python27\lib\site-packages\samcli\local\docker\lambda_container.py", line 66, in __init__
    image = LambdaContainer._get_image(image_builder, runtime, layers)
  File "c:\python27\lib\site-packages\samcli\local\docker\lambda_container.py", line 157, in _get_image
    return image_builder.build(runtime, layers)
  File "c:\python27\lib\site-packages\samcli\local\docker\lambda_image.py", line 96, in build
    downloaded_layers = self.layer_downloader.download_all(layers, self.force_image_build)
  File "c:\python27\lib\site-packages\samcli\local\layers\layer_downloader.py", line 72, in download_all
    layer_dirs.append(self.download(layer, force))
  File "c:\python27\lib\site-packages\samcli\local\layers\layer_downloader.py", line 98, in download
    layer_path = Path(self.layer_cache).joinpath(layer.name).resolve()  # pylint: disable=no-member
  File "c:\python27\lib\site-packages\pathlib2\__init__.py", line 1343, in resolve
    self.stat()
  File "c:\python27\lib\site-packages\pathlib2\__init__.py", line 1356, in stat
    return self._accessor.stat(self)
  File "c:\python27\lib\site-packages\pathlib2\__init__.py", line 541, in wrapped
    return strfunc(str(pathobj), *args)
WindowsError: [Error 2] The system cannot find the file specified: 'C:\\Users\\Administrator\\AppData\\Roaming\\AWS SAM\\layers-pkg\\demo_layer-1-1e6fe5e501'
```

### Expected result

The layer should have been pulled locally and worked as seen while testing with Python3.x

### Additional environment details

1. OS: Windows Server 2016 ( ami-01b701d1a348a0d00 - us-east-1 )
2. Package Versions:

`Python2`
`SAM: SAM CLI, version 0.11.0`
`AWS CLI: aws-cli/1.16.108 Python/2.7.16rc1 Windows/10 botocore/1.12.98`
`Docker : Docker version 18.03.1-ee-3, build b9a5c95`

`Python3`
`SAM CLI: SAM CLI, version 0.11.0`
`AWS CLI: aws-cli/1.16.108 Python/3.7.0 Windows/10 botocore/1.12.98`
`Docker : Docker version 18.03.1-ee-3, build b9a5c95`

**Comment by kwluo90** [comment]
Looks like it's broken with python 2.7 -- I was able to get it to work after switching to using the python3 based versions of aws-sam-cli and awscli.

**Comment by aaronsturm** [comment]
I am having the same issue but with node. I added the layer `arn:aws:lambda:us-west-2:764866452798:layer:libreoffice:1` and get
```
FileNotFoundError: [Errno 2] No such file or directory: '/home/aaronsturm/.aws-sam/layers-pkg/libreoffice-1-8b2813413a'
```
I know it isn't a problem with the layer as I'm able to run it in Lambda, just not locally. I've tried `sam build` and `sam build --use-container` as well as remove all of the Docker images to try and force it to download.

Am I missing something or is this a bug?
